agent/agent.go

```diff
@@ -333,11 +333,11 @@ func (a *Agent) run(ctx context.Context) {
 					a.config.SessionTracker.SessionError(err)
 				}
 
-				log.G(ctx).WithError(err).Error("agent: session failed")
 				backoff = initialSessionFailureBackoff + 2*backoff
 				if backoff > maxSessionFailureBackoff {
 					backoff = maxSessionFailureBackoff
 				}
+				log.G(ctx).WithError(err).Errorf("agent: session failed. Backoff period: %d", backoff)
```

Comment:
Use a field here: `log.G(ctx).WithError(err).WithField("backoff", backoff).Errorf("agent: session failed")`.

Note this is the maximum backoff range for a random backoff. Actual backoff is rand([0, backoff)). ;)